src/bin/cli/stats/block0/wallets.rs

```diff
@@ -0,0 +1,37 @@
+use catalyst_toolbox::stats::block0::wallets::calculate_wallet_distribution;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+pub struct WalletsCommand {
+    #[structopt(long = "support-lovelace")]
+    pub support_lovelace: bool,
+    #[structopt(long = "block0")]
+    pub block0: String,
+    #[structopt(long = "threshold")]
+    pub threshold: u64,
+    #[structopt(subcommand)]
+    pub command: Command,
+}
+
+#[derive(StructOpt, Debug)]
+pub enum Command {
+    Count,
+    Ada,
+}
+
+impl WalletsCommand {
+    pub fn exec(&self) -> Result<(), catalyst_toolbox::stats::Error> {
+        match self.command {
+            Command::Count => {
+                calculate_wallet_distribution(&self.block0, self.threshold, self.support_lovelace)?
```

Comment:
I don't see how this could differ from the calculations based on snapshot

Reply:
Well no always you have snapshot at hand. sometime it's just block0.